components/forms/select-input/select-input.js

```diff
@@ -0,0 +1,134 @@
+function SelectInput(id, options, validationMsg, placeholder, isRequired) {
+  let selectInput
+  let selectElement
+  let validationMsgElement
+  let isOpen = false
+  let displayValidationMsg = false
+  const _options = options
+  const _validationMsg = validationMsg
+  const _placeholder = isRequired ? placeholder + '*' : placeholder
+  let _valid = false
+  let _value = 'not set'
+  let _index = 0
+  let shouldPopulateList = true
+
+  const cacheDom = function() {
+    selectInput = document.getElementById(id)
+    selectElement = selectInput.querySelector('.msds-select-input__select')
+    validationMsgElement = selectInput.querySelector('.msds-select-input__validation-msg')
+  }
+
+  const bindEvents = function() {
+    selectElement.addEventListener('change', () => render())
+  }
+
+  const render = function() {
+    _index = selectElement.value
+    validationMsgElement.innerHTML = _validationMsg
```

Comment:
validationMsgElement.innerHTML = _validationMsg line can be define in cacheDom since _validationMsg always will be the same so it doesnt make sense to execute this line every time when render() is executed

Reply:
So that means you would run validationMsgElement.innerHTML = _validationMsg in cachedom? i would rather only manipulate the html in render as much as possible

Reply:
Yes. It is because _validationMsg is set only once so the value will be apply on the first execution on render method. Since value of _validationMsg is not dynamic, all next executions will assign exactly the same value so it's not really manipulating the html. But if you want to keep it there is fine - assigning variable is not a big performance impact.